tests/test_parsers.py

```diff
@@ -62,41 +54,106 @@ def test_gelfparser_parse_gzipped_file(fs, gelf_logger):
+                "{}\n",
+                # As above but raises TypeError: list indices must be integers or slices, not str
+                "[]\n",
+                # Another assumed valid gelf
+                '{"short_message": {"username": "This seems valid too."}}\n',
+            ]
+        )
+
+        file.seek(0)
+        parser = GELFParser()
+        events = list(parser.parse(file))
 
-    parser = GELFParser()
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name, chunksize=1))
-    assert len(events) == 4
-    assert events[0] == '{"username": "foo"}'
-    assert events[1] == '{"username": "bar"}'
-    assert events[2] == '{"username": "baz"}'
-    assert events[3] == '{"username": "lol"}'
-
-    parser = GELFParser()
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name, chunksize=2))
-    assert len(events) == 4
-    assert events[0] == '{"username": "foo"}'
-    assert events[1] == '{"username": "bar"}'
-    assert events[2] == '{"username": "baz"}'
-    assert events[3] == '{"username": "lol"}'
-
-    parser = GELFParser()
-    events = list(parser.parse(gelf_logger.handlers[0].stream.name, chunksize=10))
-    assert len(events) == 4
-    assert events[0] == '{"username": "foo"}'
-    assert events[1] == '{"username": "bar"}'
-    assert events[2] == '{"username": "baz"}'
-    assert events[3] == '{"username": "lol"}'
+    assert len(events) == 2
+    assert events[0] == "This seems valid."
+    assert events[1] == {"username": "This seems valid too."}
+
+    caplog.clear()
+    with StringIO() as file:
+        file.write("{ This is not valid json and raises json.decoder.JSONDecodeError\n")
+        file.seek(0)
+        parser = GELFParser()
+        with caplog.at_level(logging.DEBUG):
+            events = list(parser.parse(file))
+
+    assert len(events) == 0
+    assert caplog.record_tuples[1:] == [
+        (
+            "ralph.parsers",
+            logging.ERROR,
+            "Input event '{ This is not valid json and raises json.decoder.JSONDecodeError\n' "
+            "is not a valid JSON string! It will be ignored.",
+        ),
+        (
+            "ralph.parsers",
+            logging.DEBUG,
+            "Raised error was: Expecting property name enclosed in double quotes: "
+            "line 1 column 3 (char 2)",
+        )
+    ]
+
+    caplog.clear()
+    with StringIO() as file:
+        file.write("{}")
+        file.seek(0)
+        parser = GELFParser()
+        with caplog.at_level(logging.DEBUG):
+            events = list(parser.parse(file))
+
+    assert len(events) == 0
+    print(caplog.record_tuples)
+    assert caplog.record_tuples[1:] == [
+        (
+            "ralph.parsers",
+            logging.ERROR,
+            "Input event '{}' doesn't comply with GELF format! It will be ignored.",
+        ),
+        (
+            "ralph.parsers",
+            logging.DEBUG,
+            "Raised error was: 'short_message'",
+        )
+    ]
```

Comment:
```suggestion
    assert (
            "ralph.parsers",
            logging.ERROR,
            "Input event '{}' doesn't comply with GELF format! It will be ignored.",
        ) in caplog.record_tuples
    assert (
            "ralph.parsers",
            logging.DEBUG,
            "Raised error was: 'short_message'",
        ) in caplog.record_tuples
```